kmod/src/kernelcompat.c

```diff
@@ -67,12 +67,19 @@ kc_generic_file_buffered_write(struct kiocb *iocb, const struct iovec *iov,
 			       unsigned long nr_segs, loff_t pos, loff_t *ppos,
 			       size_t count, ssize_t written)
 {
+#ifndef KC_GENERIC_PERFORM_WRITE_KIOCB_IOV_ITER
 	struct file *file = iocb->ki_filp;
+#endif
 	ssize_t status;
 	struct iov_iter i;
 
 	iov_iter_init(&i, WRITE, iov, nr_segs, count);
+#ifdef KC_GENERIC_PERFORM_WRITE_KIOCB_IOV_ITER
+	iocb->ki_pos = *ppos;
```

Comment:
Yes, it wasn't an issue because pos and *ppos were the same values being passed through here. But pos itself is the right choice here. So I'll change it and do the kc_ wrapper.